lib/bundler/metrics.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require "time"
+require "securerandom"
+require "digest"
+
+module Bundler
+  class Metrics
+    attr_accessor :metrics_hash, :path
+
+    def initialize
+      @path = Pathname.new(File.join(Bundler.user_home.join(".bundle"), "metrics.yml"))
+      add_metrics
+    end
+
+    # sending user agent metrics over http
+    def add_metrics
```

Comment:
This method records a lot of metrics! Let's rename this to something like `record_system_info`, and only call it when Bundler is doing something big and extensive, like `install` or `outdated`.